convolution2d

alrojo opened the pull request:
explicit use "is not None" for activation_fn, normalize_fn and provide more detailed description

alrojo committed:
convolution2d

explicit use "is not None" for activation_fn, normalize_fn and provide more detailed description

tensorflow-jenkins commented:
Can one of the admins verify this patch?

danmane commented:
@sguada PTAL

sguada commented on tensorflow/contrib/layers/python/layers/layers.py:
```diff
@@ -379,9 +379,11 @@ def convolution2d(inputs,
       If greater than 1, than the a'trous convolution is applied and `stride`
       must be set to 1.
     activation_fn: activation function.
+      set to None for linear activation
```
Move to the line above.

sguada replied:
rewrite as
activation_fn: activation function, set to None to skip it and maintain a linear activation.

alrojo replied:
@sguada Done

sguada commented:
LGTM,
jenkins, test this please

alrojo committed:
Updated convolution2d with None and description

Updated convolution2d to only skip activation_fn and normalize_fn if set to None. Updated description to reflect such.

sguada commented on tensorflow/contrib/layers/python/layers/layers.py:
```diff
@@ -378,8 +378,7 @@ def convolution2d(inputs,
       must be set to 1.
-    activation_fn: activation function.
-      set to None for linear activation
+    activation_fn: activation function, set to None to skip it and maintain a linear activation.
```
Sorry try to keep the max length of each line under 80 chars.
activation_fn: activation function, set to None to skip it and maintain
  a linear activation.

alrojo replied:
@sguada updated

sguada commented:
@danmane do you know why Jenkins is not running?

alrojo committed:
update according to PEP8

sguada commented:
@alrojo would you mind updating the other layers that have activation_fn and normalizer_fn too?

sguada commented:
@tensorflow-jenkins test this please

alrojo commented:
@sguada I'll look into the other layers, why did the test fail?

danmane commented:
It looks like the test failure was a flake, since it was a timeout in code clearly unrelated to your change. I would consider that a clean build, but I will re-run the tests anyway.
@tensorflow-jenkins test this please.

alrojo committed:
normalizer_fn and activation_fn updated

updated documentation and rejection type for all layers

alrojo committed:
removing space

alrojo commented:
@sguada updated all of the other layers with activation_fn and normalizer_fn. I also updated a typo in the docs: normalize_fn -> normalizer_fn

sguada commented:
Thanks @alrojo 

@tensorflow-jenkins test this please

martinwicke commented:
Merged, the remaining test failure was a flake.
